controllers/hostingdeploymentautoscalingjob/hostingdeploymentautoscalingjob_controller.go

```diff
@@ -0,0 +1,455 @@
+	// TODO this is not being used at the moment, delete if not required
+	RegisteredTargetsJobStatus = "RegisteredTargets"
+
+	// Could have failed either at step1 or step2
+	FailedAutoscalingJobStatus = "FailedAutoscalingJob"
+
+	// This Status will likely not show up, is it needed
+	DeletedAutoscalingJobStatus = "DeletedAutoscalingJob"
+
+	// https://docs.aws.amazon.com/autoscaling/application/APIReference/API_ScalingPolicy.html
+	MaxPolicyNameLength = 256
+
+	// Default values for Autoscaling in the SageMaker Service
+	ScalableDimension            = "sagemaker:variant:DesiredInstanceCount"
+	PolicyType                   = "TargetTrackingScaling"
+	DefaultAutoscalingPolicyName = "SageMakerEndpointInvocationScalingPolicy"
+)
+
+// Reconciler reconciles a HostingDeploymentAutoscalingJob object
+type Reconciler struct {
+	client.Client
+	Log                                logr.Logger
+	createApplicationAutoscalingClient clientwrapper.ApplicationAutoscalingClientWrapperProvider
+	awsConfigLoader                    controllers.AwsConfigLoader
+}
+
+// NewHostingDeploymentAutoscalingJobReconciler creates a new reconciler with the default ApplicationAutoscaling client.
+func NewHostingDeploymentAutoscalingJobReconciler(client client.Client, log logr.Logger, pollInterval time.Duration) *Reconciler {
+	return &Reconciler{
+		Client: client,
+		Log:    log,
+		// TODO: Model calls ClientAPI, check
+		createApplicationAutoscalingClient: func(cfg aws.Config) clientwrapper.ApplicationAutoscalingClientWrapper {
+			return clientwrapper.NewApplicationAutoscalingClientWrapper(applicationautoscaling.New(cfg))
+		},
+		awsConfigLoader: controllers.NewAwsConfigLoader(),
+	}
+}
+
+// +kubebuilder:rbac:groups=sagemaker.aws.amazon.com,resources=hostingdeploymentautoscalingjobs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=sagemaker.aws.amazon.com,resources=hostingdeploymentautoscalingjobs/status,verbs=get;update;patch
+
+// Reconcile attempts to reconcile the SageMaker resource state with the k8s desired state.
+// Review: Clean this up
+func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := reconcileRequestContext{
+		Context:                         context.Background(),
+		Log:                             r.Log.WithValues("hostingdeploymentautoscalingjob", req.NamespacedName),
+		HostingDeploymentAutoscalingJob: new(hostingdeploymentautoscalingjobv1.HostingDeploymentAutoscalingJob),
+	}
+
+	ctx.Log.Info("Getting resource")
+
+	// Get state from etcd
+	if err := r.Get(ctx, req.NamespacedName, ctx.HostingDeploymentAutoscalingJob); err != nil {
+		ctx.Log.Info("Unable to fetch HostingDeploymentAutoscalingJob", "reason", err)
+		if apierrs.IsNotFound(err) {
+			return controllers.NoRequeue()
+		}
+		return controllers.RequeueImmediately()
+	}
+
+	if err := r.reconcileHostingDeploymentAutoscalingJob(ctx); err != nil {
+		ctx.Log.Info("Got error while reconciling, will retry", "err", err)
```

Comment:
your comment and next two lines does not match..check with model or hosting deployment controller and you should do exactly like that

Reply:
Ack

Reply:
or at least your function `reconcileHostingDeploymentAutoscalingJob` semantic should match to them

Reply:
Done, not exactly same as model, but similar